apps/vc-api/src/askar/askar.config.ts

```diff
@@ -0,0 +1,91 @@
+import { AskarModule, AskarWallet } from "@credo-ts/askar";
+import { Agent, InitConfig, SigningProviderRegistry, ConsoleLogger, WalletConfig } from "@credo-ts/core";
+import { agentDependencies } from "@credo-ts/node";
+import { ariesAskar } from "@hyperledger/aries-askar-nodejs";
+import { Injectable, OnModuleInit, OnModuleDestroy } from '@nestjs/common';
+
+@Injectable()
+export class AskarService implements OnModuleInit, OnModuleDestroy{
+  private readonly walletConfig: WalletConfig = {
+    id: process.env['ASKAR_WALLET_ID'],
+    key: process.env['ASKAR_WALLET_KEY'],
+    storage: {
+      type: process.env['ASKAR_WALLET_DB_TYPE'],
+      // config: {
+      //   inMemory: true,
+      // },
+    },
+  };
+  private readonly agent: Agent<{ askar: AskarModule; }>;
+  private readonly wallet: AskarWallet;
+  private initialized: boolean = false;
+
+  constructor() {
+    // Initialize agent and wallet synchronously
+    const config: InitConfig = {
+      label: process.env['ASKAR_LABEL'],
+      walletConfig: this.walletConfig,
+    };
+
+    // Create the agent
+    this.agent = new Agent({
+      config,
+      dependencies: agentDependencies,
+      modules: {
+        askar: new AskarModule({
+          ariesAskar,
+        }),
+      },
+    });
+
+    // Create the wallet
+    this.wallet = new AskarWallet(
+      new ConsoleLogger(),
+      new agentDependencies.FileSystem(),
+      new SigningProviderRegistry([]),
+    );
+  }
+
+  async onModuleInit() {
+    // This method is called after the module is initialized
+    await this.initialize();
+  }
+
+  // initializes the askar agent and wallet for operations
+  private async initialize() {
+    if (!this.agent.isInitialized) {
+      await this.agent.initialize();
+    }
+    await this.wallet.open(this.walletConfig);
+    this.initialized = true;
+  }
+
+  // Accessor for the agent
+  public getAskarAgent(): Agent<{ askar: AskarModule }> {
```

Comment:
I think it is better to think of it as a Credo agent. However, assuming this class is renamed to `CredoService`, I think having `Credo` in the method name would be redundant
```suggestion
  public getAgent(): Agent<{ askar: AskarModule }> {
```